src/main/java/ch/njol/skript/classes/data/BukkitClasses.java

```diff
@@ -732,17 +733,19 @@ public String toVariableNameString(InventoryType o) {
 					@Override
 					@Nullable
 					public Player parse(String s, ParseContext context) {
-						if (context == ParseContext.COMMAND) {
-							if (UUID_PATTERN.matcher(s).matches())
-								return Bukkit.getPlayer(UUID.fromString(s));
-							List<Player> ps = Bukkit.matchPlayer(s);
-							if (ps.size() == 1)
-								return ps.get(0);
-							if (ps.size() == 0)
-								Skript.error(String.format(Language.get("commands.no player starts with"), s));
-							else
-								Skript.error(String.format(Language.get("commands.multiple players start with"), s));
-							return null;
+						if (!s.isEmpty()) {
+							if (context == ParseContext.COMMAND) {
+								if (UUID_PATTERN.matcher(s).matches())
+									return Bukkit.getPlayer(UUID.fromString(s));
+								List<Player> ps = Bukkit.matchPlayer(s);
+								if (ps.size() == 1)
+									return ps.get(0);
+								if (ps.size() == 0)
+									Skript.error(String.format(Language.get("commands.no player starts with"), s));
+								else
+									Skript.error(String.format(Language.get("commands.multiple players start with"), s));
+								return null;
+							}
 						}
 						assert false;
 						return null;
```

Comment:
This assert should only be called if context is not COMMAND, in no other cases

Reply:
Actually, if you look at the ``canParse`` method, it allows parsing only if the context is a command. Should I fix the whole type?

Reply:
Yes that's why there's a false assert, because the argument may never be any other value

Reply:
I wish I could, but it's something I'm not sure if I'm capable of

Reply:
If I understood the RC correctly I think it's just like this, can you confirm @TPGamesNL ?
```suggestion
						if (!s.isEmpty()) {
							if (context == ParseContext.COMMAND) {
								if (UUID_PATTERN.matcher(s).matches())
									return Bukkit.getPlayer(UUID.fromString(s));
								List<Player> ps = Bukkit.matchPlayer(s);
								if (ps.size() == 1)
									return ps.get(0);
								if (ps.size() == 0)
									Skript.error(String.format(Language.get("commands.no player starts with"), s));
								else
									Skript.error(String.format(Language.get("commands.multiple players start with"), s));
								return null;
							}
						}
						
						if (context != ParseContext.COMMAND)
							assert false;
						
						return null;
```

Reply:
I would prefer
```suggestion
						
						if (context == ParseContext.COMMAND) {
							if (s.isEmpty())
								return null;
							if (UUID_PATTERN.matcher(s).matches())
								return Bukkit.getPlayer(UUID.fromString(s));
							List<Player> ps = Bukkit.matchPlayer(s);
							if (ps.size() == 1)
								return ps.get(0);
							if (ps.size() == 0)
								Skript.error(String.format(Language.get("commands.no player starts with"), s));
							else
								Skript.error(String.format(Language.get("commands.multiple players start with"), s));
							return null;
						}
						assert false;
						return null;
```